### PR TITLE
Feat33 prognose view

### PR DESCRIPTION
**Pull request by Enyuuus**
# Description

a customer-friendly view for forecast has been made and the create and change fucntionality of forecast works. also the way to nav to standards has changed.
add the standard view and it is now posible toe change the standards

# Checklist:

- [x] My code runs
- [x] My code follows the coding guidelines
- [x] I rebased my branch on development
- [x] My code only has comments when necessary for understandability 
- [x] I removed commented out code
- [x] My code corresponds with the documentation
- [x] I have updated the documentation when changes were needed
- [x] All unit-tests succeeded (Except for not implemented exceptions)

# Optional:

- [x] Is the branch linked to an issue?

### CODE REVIEW

File: Bumbodium.Data/Repositories/ForecastRepo.cs [hunk 1]
```diff
@@ -27,18 +27,18 @@ public ForecastRepo(BumbodiumContext ctx, DepartmentRepo departmentRepo)
...
             //TODO Make Country relative to the forecast
             _standards = _ctx.Standards.Where(s => s.Country == Country.Netherlands).ToList();
             var weekprognose = WeekCalEmployes(forecasts);
-            foreach(var prognosis in weekprognose)
+            foreach (var prognosis in weekprognose)
                 _ctx.Forecast.Add(prognosis);
             _ctx.SaveChanges();
 
-        }
+        }*/
```
Comment (Martijn1030) [inline Bumbodium.Data/Repositories/ForecastRepo.cs:41]: Commented out code ? 
rock-sus

Comment (Enyuuus) [reply Bumbodium.Data/Repositories/ForecastRepo.cs]: fixed

File: Bumbodium/Views/Forecast/ChangeInput.cshtml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                        <li class="list-group-item mb-2"><h3 class="m-2">@Model.CountTotaalHoursPerDay(forecastDay.Key)</h3></li>
+                    </ul>
+                </div>
+            }
+
+        </div>
+        <br />
+        <div class="row pb-0">
+            <div class="col text-center p-0 ">
+                <button type="submit" class="btn btn-lg btn-primary btn-outline-warning">Maakt een nieuwe prognose aan</button>
```
Comment (Martijn1030) [inline Bumbodium/Views/Forecast/ChangeInput.cshtml:44]: Maakt een nieuwe prognose aan is beetje vaag, Toevoegen of Aanmaken leest fijner denk ik

File: Bumbodium/Views/Forecast/ChangeOutput.cshtml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                        }
+                    </ul>
+                </div>
+            }
+        </div>
+        <br />
+        <div class="row pb-0">
+            <div class="row pb-0">
+                <div class="col text-center p-0">
+                    <button type="submit" class="btn btn-lg btn-primary btn-outline-warning">Pas de bestaande prognose aan</button>
```
Comment (Martijn1030) [inline Bumbodium/Views/Forecast/ChangeOutput.cshtml:64]: Aanpassen of Toepassen of opslaan ofz

File: Bumbodium/Views/Forecast/Index.cshtml [hunk 4]
```diff
@@ -1,57 +1,69 @@
...
+            </div>
+        }
+        <br />
+    </div>
+    <br />
+    <div class="row pb-5">
+        <div class="col text-center p-0 ">
+            @if (!Model.WeekBeforeNow)
+            {
+                <a asp-action="ChangeInput" class="btn btn-lg btn-primary btn-outline-warning" asp-route-id="@calanderVal">Verander de prognose dag</a>
```
Comment (Martijn1030) [inline Bumbodium/Views/Forecast/Index.cshtml:50]: Input prognose aanpassen

File: Bumbodium/Views/Forecast/Index.cshtml [hunk 5]
```diff
@@ -1,57 +1,69 @@
...
+            }
+            else
+            {
+                <p class="btn btn-lg btn-outline-secondary">Verander de prognose dag</p>
+            }
+        </div>
+        <div class="col text-center p-0">
+            @if (!Model.WeekBeforeNow)
+            {
+                <a asp-action="ChangeOutput" class="btn btn-lg btn-primary btn-outline-warning" asp-route-id="@calanderVal">Veranderen de uitkomst velden</a>
```
Comment (Martijn1030) [inline Bumbodium/Views/Forecast/Index.cshtml:60]: Output aanpassen

File: Bumbodium/Views/Standard/ChangeStandards.cshtml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                    <input hidden asp-for="Standards[i].Subject" />
+                    <input asp-for="Standards[i].Value" class="m-3 text-lg text-center" />
+                    <span asp-validation-for="Standards[i].Value" class="text-danger"></span>
+                </div>
+            </div>
+        }
+        <br />
+        <div class="row pb-0">
+            <div class="col text-center p-0 ">
+                <button type="submit" class="btn btn-lg btn-primary btn-outline-warning">Save de veranderde Normeringen</button>
```
Comment (Martijn1030) [inline Bumbodium/Views/Standard/ChangeStandards.cshtml:36]: Aanpassingen opslaan of Aanpassen of Toepassen

File: Bumbodium/Views/Standard/Index.cshtml [hunk 7]
```diff
@@ -1,47 +1,55 @@
...
-        <option value="default"> --Select land--</option>
-        @foreach (var coutnry in ViewBag.Countrys)
-        {
-            <option value="@coutnry.ToString()">@coutnry.ToString()</option>
-        }
+<form asp-controller="Standard" asp-action="Index">
+
+    <label class="text-lg">Kies een land: </label>
+    <select asp-for="Country" asp-items="Html.GetEnumSelectList<Country>()" onchange="this.form.submit()">
+        <option selected="selected" value="" disabled>Please select</option>
```
Comment (Martijn1030) [inline Bumbodium/Views/Standard/Index.cshtml:15]: Selecteer...

**Review by Martijn1030 — CHANGES_REQUESTED** [review]
Hier en daar nog wat text aanpassen, verder is het goed

File: Bumbodium/Views/Forecast/ChangeInput.cshtml (hunk 2 above)
Comment (Bewerkt) [reply Bumbodium/Views/Forecast/ChangeInput.cshtml:44]: Eens

File: Bumbodium/Views/Forecast/Index.cshtml [hunk 8]
```diff
@@ -1,57 +1,69 @@
...
+    <br />
+    <div class="row pb-5">
+        <div class="col text-center p-0 ">
+            @if (!Model.WeekBeforeNow)
+            {
+                <a asp-action="ChangeInput" class="btn btn-lg btn-primary btn-outline-warning" asp-route-id="@calanderVal">Verander de prognose dag</a>
+            }
+            else
+            {
+                <p class="btn btn-lg btn-outline-secondary">Verander de prognose dag</p>
```
Comment (Bewerkt) [inline Bumbodium/Views/Forecast/Index.cshtml:54]: Prognose dag aanpassen/wijzigen

**Review by Bewerkt — CHANGES_REQUESTED** [review]
Het spijt me als ik deleted lines heb verbetert in taal

File: Bumbodium/Views/Forecast/ChangeOutput.cshtml [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+<h1>Prognose</h1>
+<hr>
+<br />
+<form action="/Forecast/ChangeOutput" method="post">
+    <input hidden asp-for="StartOfWeekDate" />
+    <div class="container">
+        <div class="row shadow-lg">
+            <div class="col-auto p-0 bg-white">
+                <ul class="list-group-flush p-0">
+                    <li class="list-group-item bg-black text-black"><h3 class="bg-black text-light p-1"> Year: @Model.StartOfWeekDate.Year </h3></li>
```
Comment (Bewerkt) [inline Bumbodium/Views/Forecast/ChangeOutput.cshtml:12]: Jaar:
Nederlands gebruiken

File: Bumbodium/Views/Forecast/ChangeOutput.cshtml [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+                        }
+                    </ul>
+                </div>
+            }
+        </div>
+        <br />
+        <div class="row pb-0">
+            <div class="row pb-0">
+                <div class="col text-center p-0">
+                    <button type="submit" class="btn btn-lg btn-primary btn-outline-warning">Aanpassingen oplsaan</button>
```
Comment (Bewerkt) [inline Bumbodium/Views/Forecast/ChangeOutput.cshtml:64]: opslaan

File: Bumbodium/Views/Standard/ChangeStandards.cshtml [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+                    <input hidden asp-for="Standards[i].Subject" />
+                    <input asp-for="Standards[i].Value" class="m-3 text-lg text-center" />
+                    <span asp-validation-for="Standards[i].Value" class="text-danger"></span>
+                </div>
+            </div>
+        }
+        <br />
+        <div class="row pb-0">
+            <div class="col text-center p-0 ">
+                <button type="submit" class="btn btn-lg btn-primary btn-outline-warning">Opslaan de veranderde Normeringen</button>
```
Comment (Bewerkt) [inline Bumbodium/Views/Standard/ChangeStandards.cshtml:36]: Veranderde normeringen opslaan

File: Bumbodium/Views/Standard/ChangeStandards.cshtml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+<h1>Normeringen</h1>
+<hr>
+<br />
+
+<form asp-action="ChangeStandards" method="post">
+    <input hidden asp-for="Country" />
+    <div class="container justify-content-center">
+        <div class="row shadow-lg m-0">
+            <div class="col m-0 p-0 bg-black ">
+                <h2 class="p-3 text-lg text-light">Normeringen van het land @Model.Country</h2>
```
Comment (Bewerkt) [inline Bumbodium/Views/Standard/ChangeStandards.cshtml:16]: 'het land' mag je weglaten wat mij betreft

File: Bumbodium/Views/Standard/Index.cshtml [hunk 13]
```diff
@@ -1,47 +1,55 @@
...
-else
-{
-    <h1>Dit land heeft nog geen normeringen</h1>
-}
\ No newline at end of file
+
+<div class="container justify-content-center">
+    <div class="row shadow-lg m-0">
+        <div class="col m-0 p-0 bg-black ">
+            <h2 class="p-3 text-lg text-light">Normeringen van het land @Model.Country</h2>
```
Comment (Bewerkt) [inline Bumbodium/Views/Standard/Index.cshtml:25]: 'het land' mag weg

**Review by Bewerkt — CHANGES_REQUESTED** [review]
paar dingetjes

**Review by Bewerkt — APPROVED** [review]
Ok.

**Review by Martijn1030 — APPROVED** [review]

**Pull request merged**